components/uniset/src/uniset.rs

```diff
@@ -0,0 +1,328 @@
+    // Allows for traits of fixed size arrays
+    inv_list: Vec<u32>,
+}
+
+impl UnicodeSet {
+    /// Returns Result of UnicodeSet from serialized string
+    ///
+    /// Returns an error if the serialized string fails to parse.
+    /// The serialized string requires starting capacity integer, followed by space delimited
+    /// integer code points. There must be an even number of elements (not including the
+    /// capacity int), and must be in ascending sorted order.
+    ///
+    /// Example String: `"4 0 5 10 15"` designates a capacity of size `4`, followed by 2 ranges
+    /// The ranges are `{0, 4}` and `{10, 14}` inclusive
+    pub fn new(serialize: &str) -> Result<UnicodeSet, Box<dyn Error>> {
+        match parse_serial_string(serialize) {
+            Ok(serialize) => Ok(UnicodeSet {
+                inv_list: serialize,
+            }),
+            Err(e) => Err(e),
+        }
+    }
+
+    /// Returns Result of UnicodeSet from a single pair of integers defining a range
+    ///
+    /// `start`: inclusive, `end`: exclusive
+    ///
+    /// Returns an error if the range is invalid (out of order and out of bounds).
+    ///
+    /// Example Call: `UnicodeSet::from_range(&0, &15)`
+    pub fn from_range(start: &u32, end: &u32) -> Result<UnicodeSet, Box<dyn Error>> {
+        if start > end {
+            return Err("Range is out of order".into());
+        }
+        if start < &CODEPOINT_MIN || end > &CODEPOINT_MAX {
+            return Err("Range is out of bounds".into());
+        }
+        Ok(UnicodeSet {
+            inv_list: vec![*start, *end],
+        })
+    }
+
+    /// Returns UnicodeSet spanning entire Unicode range
+    ///
+    /// The range spans from `0x0 -> 0x10FFFF` inclusive
+    pub fn all() -> UnicodeSet {
+        UnicodeSet {
+            inv_list: vec![CODEPOINT_MIN, CODEPOINT_MAX + 1],
+        }
+    }
+
+    /// Returns UnicodeSet spanning BMP range
+    ///
+    /// The range spans from `0x0 -> 0xFFFF` inclusive
+    pub fn bmp() -> UnicodeSet {
+        UnicodeSet {
+            inv_list: vec![CODEPOINT_MIN, BMP_MAX + 1],
+        }
+    }
+
+    /// Returns an `Iter` of start and stop `u32` points of the UnicodeSet
+    pub fn iter(&self) -> Iter<u32> {
+        self.inv_list.iter()
+    }
+
+    /// Returns the cardinality of the UnicodeSet
+    ///
+    ///
+    pub fn size(&self) -> Result<usize, Box<dyn Error>> {
+        if self.inv_list.len() < 2 {
+            return Err("UnicodeSet length < 2".into());
+        }
+        let end_point_sum: u32 = self.iter().skip(1).step_by(2).sum::<u32>();
+        let start_point_sum: u32 = self.iter().step_by(2).sum::<u32>();
+        Ok((end_point_sum - start_point_sum) as usize)
+    }
+
+    /// Returns whether or not the UnicodeSet is empty
+    pub fn is_empty(&self) -> bool {
+        self.inv_list.len() < 2 // unsure if this is appropriate definition of just self.inv_list.is_empty()
+    }
+
+    /// Wrapper for contains conditions closures
```

Comment:
I agree to the former so I included an example, but this will only matter for developers as the `contains` method is not public (rustdoc will not expose it on the HTML page).

Regarding the latter, here `std::vec::Vec::binary_search` [requires a reference](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.binary_search). I will change this for the public API contains methods though for consistency sake.